src/cls/IPM/ResourceProcessor/Abstract.cls

```diff
@@ -192,6 +192,13 @@ Method OnPhase(pPhase As %String, ByRef pParams, Output pResourceHandled As %Boo
 	Quit $$$OK
 }
 
+/// Some processors may need to do something custom when a custom phase is executed.
+/// Such as the CPF processor, which optionally merges a CPF file during a custom phase.
+Method OnCustomPhase(pCustomPhase As %String, ByRef pParams) As %Status
```

Comment:
I like this abstraction! However, I think we may want to put this in a another intermediate subclass for non-document processor classes. Document processors always execute during regular phases since they don't do things in just one phase (need to do things across reload and compile at a minimum)